core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -868,11 +872,24 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
      * <p>
      * If the snapshot is still running cancels the snapshot first and then deletes it from the repository.
      *
-     * @param snapshotId snapshot id
+     * @param snapshot   snapshot
      * @param listener   listener
      */
+    public void deleteSnapshot(final Snapshot snapshot, final DeleteSnapshotListener listener) {
+        deleteSnapshot(resolveSnapshotId(snapshot.getRepository(), snapshot.getName()), listener);
+    }
+
+    /**
+     * Deletes snapshot from repository.
+     * <p>
+     * If the snapshot is still running cancels the snapshot first and then deletes it from the repository.
+     *
+     * @param snapshotId  snapshot id
+     * @param listener    listener
+     */
     public void deleteSnapshot(final SnapshotId snapshotId, final DeleteSnapshotListener listener) {
-        validate(snapshotId);
+        final Snapshot snapshot = snapshotId.getSnapshot();
+        validate(snapshot);
```

Comment:
no need to validate? can't we assume that once we have a SnapshotId that validation has happened?